src/AddressSpace.cc

```diff
@@ -373,22 +376,61 @@ static const AddressSpace::SyscallType entry_points[] = {
     AddressSpace::RECORDING_ONLY },
 };
 
-static remote_code_ptr entry_ip_from_index(size_t i) {
-  return remote_code_ptr(RR_PAGE_ADDR + RR_PAGE_SYSCALL_STUB_SIZE * i);
+static int rr_page_syscall_stub_size(SupportedArch arch) {
+  int val = 0;
+  switch (arch) {
+    case x86:
+    case x86_64:
+      val = 3;
+      break;
+    case aarch64:
+      val = 8;
+      break;
+    default:
+      FATAL() << "Syscall stub size not defined for this architecture";
+  }
+  if (arch == NativeArch::arch()) {
+    DEBUG_ASSERT(val == RR_PAGE_SYSCALL_STUB_SIZE);
```

Comment:
What's the benefit of keeping this constant and the assertion?

Reply:
The syscallbuf code makes use of the constant to determine the addresses it needs. The assertion ensures that the constant defined in preload_interface is consistent with the enum values defined here (at least for the current architecture, but it ever gets out of sync, you'd notice before it becomes a stranger problem assuming you're on the problematic architecture).